utils/nx-plugin/clone-build-targets-for-release.cjs

```diff
@@ -0,0 +1,130 @@
+ */
+function cloneTarget(projectTargets, targetName) {
+  const dependsOn = projectTargets[targetName].dependsOn;
+  const dependenciesToClone = dependsOn
+    ? dependsOn.filter(
+        (targetDependency) => !isParentDependency(targetDependency)
+      )
+    : [];
+  /** @type {Record<string, TargetConfiguration>} */
+  const clonedDependencies = mergeObjects(
+    ...dependenciesToClone.map((targetDependency) => {
+      const targetDependencyName =
+        typeof targetDependency === 'string'
+          ? targetDependency
+          : targetDependency.target;
+      if (!(targetDependencyName in projectTargets)) {
+        return {};
+      }
+      const {target: clonedDependency, localDependencies: deepDependencies} =
+        cloneTarget(projectTargets, targetDependencyName);
+      return {...deepDependencies, [targetDependencyName]: clonedDependency};
+    })
+  );
+  return {
+    target: JSON.parse(JSON.stringify(projectTargets[targetName])),
+    localDependencies: clonedDependencies,
+  };
+}
+
+/**
+ * @param {TargetConfiguration} target
+ * @param {string} dependencyToReplace E.g.: `build` or `^build`
+ * @param {string[]} newDependencies
+ */
+function replaceDependency(target, dependencyToReplace, newDependencies) {
+  if (!target.dependsOn) {
+    return;
+  }
+  target.dependsOn = target.dependsOn.flatMap((dependency) => {
+    if (typeof dependency === 'string') {
+      if (dependency === dependencyToReplace) {
+        return newDependencies;
+      }
+      return [/** @type {string | TargetDependencyConfig} */ (dependency)];
+    }
+    const targetToReplace = dependencyToReplace.startsWith('^')
+      ? dependencyToReplace.slice(1)
+      : dependencyToReplace;
+    if (dependency.target === targetToReplace) {
+      return newDependencies;
+    }
+    return [/** @type {string | TargetDependencyConfig} */ (dependency)];
+  });
+}
+
+/**
+ * @param {Record<string, TargetConfiguration>} targets
+ * @param {string} oldName
+ * @param {string} newName
+ */
+function renameTarget(targets, oldName, newName) {
+  targets[newName] = targets[oldName];
+  delete targets[oldName];
+  for (const target of Object.values(targets)) {
+    // We only cloned dependencies which are defined locally. We don't need to replace parent dependencies.
+    replaceDependency(target, oldName, [newName]);
+  }
+}
+
+/**
+ * @param {ProjectGraphProjectNode} project
+ */
+function addReleaseBuildTargets(project) {
+  const projectTargets = project.data.targets || {};
+  if (!('build' in projectTargets)) {
+    return;
+  }
+  const {target: buildTarget, localDependencies: buildTargetDependencies} =
+    cloneTarget(projectTargets, 'build');
+  const newTargets = {...buildTargetDependencies, 'release:build': buildTarget};
+  for (const dependencyName of Object.keys(buildTargetDependencies)) {
+    renameTarget(newTargets, dependencyName, `release:build:${dependencyName}`);
+  }
```

Comment:
We want to rename the new targets to avoid overwriting the old ones. In this case, I'm just prefixing them with `release:build:`.

This function will also rename them in the `dependsOn` field of other targets.